packages/ui/src/components/StyledTab.tsx

```diff
@@ -14,54 +20,76 @@ type StyledTabButtonType = StyledTabProps & {
   isSelected: boolean;
 };
 
+type StyledTabTheme = "default" | "primary";
+
+const ThemeContext = createContext<StyledTabTheme>("default");
+
+type ListProps = PropsWithChildren<{
+  theme?: StyledTabTheme;
+  stretch?: boolean;
+}>;
+
 type StyledTabType = FC<StyledTabProps> & {
   TabStyle: string;
-  List: FC<{ children: ReactNode }>;
+  List: FC<ListProps>;
+  ListDiv: FC<ListProps>;
   Button: FC<StyledTabButtonType>;
-  ListDiv: FC<{ children: ReactNode }>;
-  Group: typeof Tab.Group;
+  Group: typeof TabGroup;
   Panels: typeof Tab.Panels;
   Panel: typeof Tab.Panel;
 };
 
 const tabStyle =
-  "group flex rounded-md focus:outline-none focus-visible:ring-offset-gray-100 hover:bg-gray-300 mx-px cursor-pointer";
+  "group flex flex-1 rounded-md focus:outline-none focus-visible:ring-offset-gray-100 hover:bg-gray-300 mx-px cursor-pointer";
 
 const StyledTabInner: FC<StyledTabButtonType> = ({
   name,
   isSelected,
   icon: Icon,
-}) => (
-  <>
+}) => {
+  const theme = useContext(ThemeContext);
+
+  return (
     <span
       className={clsx(
-        "flex items-center rounded-md p-1 pl-2.5 pr-2.5 text-sm font-medium",
-        isSelected && "bg-white shadow-sm ring-1 ring-black ring-opacity-5"
+        "flex flex-1 items-center justify-center rounded-md p-1 pl-2.5 pr-2.5 text-sm font-medium",
+        isSelected && [
+          theme === "default" &&
+            "bg-white shadow-sm ring-1 ring-black ring-opacity-5",
+          theme === "primary" &&
+            "bg-blue-500 text-white shadow-sm ring-1 ring-black ring-opacity-5",
```

Comment:
It's awkward that the primary button is green but primary tabs are blue.

Should tabs also be green?

We could also add blue buttons, though I'd prefer if we had clear understanding & documentation about the difference on semantics level between green and blue if we do that (e.g. "green=primary, single button on the page, blue=secondary but important actions, gray=everything else that needs to be deemphasized").